### ISSUE TITLE
Parse error with typed expression splice in let statement

### ISSUE THREAD

**Issue by pythonissam** [ndmitchell/hlint#493]
```haskell
{-# LANGUAGE TemplateHaskell #-}

module N where

import Language.Haskell.TH

n :: Q Exp
n = [| 10 |]

nn :: Q (TExp Int)
nn = [|| 10 ||]
```

```haskell
#!/usr/bin/env stack
-- stack script --resolver lts-11.10 --package template-haskell

{-# LANGUAGE TemplateHaskell #-}

import Language.Haskell.TH
import N

main :: IO ()
main = do
  print $n
  
  -- This works
  -- print $$nn

  -- This also works
  -- nn' <- return $$nn
  -- print nn'

  -- But this fails
  let nn' = $$nn
  print nn'
```

**Comment by ndmitchell** [comment]
Thanks for the reports, they're already upstream in HSE at https://github.com/haskell-suite/haskell-src-exts/issues/278 and https://github.com/haskell-suite/haskell-src-exts/issues/383. Unfortunately HSE doesn't yet support typed quotes. I have a few issues in HLint tracking this, so will close this ticket.
